### ISSUE TITLE
CGLib proxy of spring-boot 2 is not supported

### ISSUE THREAD

**Issue by agoston** [lukas-krecan/ShedLock#162]
Hi!

Great library, I've been using `2.5.0` with spring-boot `2.1.7` and the jdbc lock provider and the following setup:

```java
@EnableScheduling
@EnableSchedulerLock(defaultLockAtMostFor = "PT12H", defaultLockAtLeastFor = "PT10S", mode = InterceptMode.PROXY_METHOD)
```

I'm encountering the following error:

```java
[...]
Caused by: org.springframework.beans.factory.BeanNotOfRequiredTypeException: Bean named 'publishIntakeCheckJob' is expected to be of type 'com.bol.amnesia.jobs.PublishIntakeCheckJob' but was actually of type 'com.sun.proxy.$Proxy207'
[...]
```

The offending line is:

```java
    @Autowired PublishIntakeCheckJob publishIntakeCheckJob;
```

which goes like this:

```java
public class PublishIntakeCheckJob implements ScheduledJob {
[...]
    @Scheduled(cron = "${intake.cron}")
    @SchedulerLock(name = JOB_NAME)
    public void execute() {}
[...]
}
```

If I change the wiring of the `publishIntakeCheckJob`:

```java
    @Autowired ScheduledJob publishIntakeCheckJob;
```

it works.

Looking at the source code, it seems that only supports jdk proxies. Since spring-boot 2, the default proxy method is cglib, for the exact reason that jdk proxies would implement the interface of the bean only, even if said interface is only used as a trait, not an old-OO-style interface.

See `AopAutoConfiguration` in spring-boot 2 for more details.

Is it possible to add some cglib support? As it stands, I'll have to add an aspect myself that would call the locks manually, which makes this nice lean library more painful to use.

**Comment by lukas-krecan** [comment]
Yes, I have been thinking about it. Will try to implement it in next few weeks. In the meantime, you can try this https://github.com/lukas-krecan/ShedLock#scheduled-method-proxy

**Comment by lukas-krecan** [comment]
Will add new config option PROXY_SCHEDULER_CGLIB. Will make it default option after it's tested in the field for a while.

**Comment by agoston** [comment]
Hey, thanks for picking it up so fast!

**Comment by lukas-krecan** [comment]
Released as 3.0.0, should take settings from Spring Boot, so no manual configuration is necessary
